### ISSUE TITLE
Stack cells

### ISSUE THREAD

**Issue by aspyatkin** [schachmat/wego#7]
I suppose cells should be stacked when terminal width is not sufficient for 4 cells of each day. Now in my terminal with 90 symbols width the output is a real mess.
![notstacked](https://cloud.githubusercontent.com/assets/4273812/7631485/f98917ea-fa53-11e4-8ec0-1d2a911f1e5b.png)

**Comment by schachmat** [comment]
Correct, that is a problem. I have not researched how to reliably and portably detect terminal width in go yet. If you have some hints/ideas, let me know.

**Comment by certifiedloud** [comment]
+1

**Comment by BrandonRomano** [comment]
@schachmat, here's a quick program showing how you can detect terminal width.

You might have to run `go get golang.org/x/crypto/ssh/terminal` before running if you haven't already got this installed.

```go
package main

import (
    "fmt"
    "golang.org/x/crypto/ssh/terminal"
)

func main() {
    width, _, _ := terminal.GetSize(0)
    fmt.Println("Width:", width)
}
```

Outputs for me:

```
Width: 159
```

**Comment by philadams** [comment]
:+1:

**Comment by kpashka** [comment]
@schachmat Also, take a look at this package - [tablewriter](https://github.com/olekukonko/tablewriter)

**Comment by schachmat** [comment]
tablewriter does not seem to support correct connections for the vertical and horizontal lines. Also embedding the date between the „Noon“ and „Evening“ columns can't be done.

**Comment by giuscri** [comment]
Apparently you can elegantly solve the problem using the core package `syscall` and doing the *ioctl* call— but check the parameter you pass to it for portability.

Check here
> http://stackoverflow.com/a/16576712/2219670
and here
> http://unix.stackexchange.com/a/93174/47999

**Comment by philcali** [comment]
I don't know how compatible the renderings are, but you might be able to accomplish this with [TermUI](https://github.com/gizak/termui)?

**Comment by marcofeltmann** [comment]
The package `golang.org/x/crypto/ssh/terminal` is deprecated.
It moved to `golang.org/x/term` instead.

Sample:
```golang
package main

import (
	"log/slog"
	"os"

	"golang.org/x/term"
)

func main() {
	ttyFd := int(os.Stdin.Fd())
	if !term.IsTerminal(ttyFd) {
		slog.Default().Error("Current Stdin File Descriptor is no terminal, aborting")
		os.Exit(1)
	}
	width, height, err := term.GetSize(ttyFd)
	if err != nil {
		slog.Default().Error(
			"Unable to get terminal size, aborting.",
			slog.Any("error", err),
		)
		os.Exit(1)
	}

	slog.Default().Info(
		"Detected terminal size.",
		slog.Int("columns", width),
		slog.Int("rows", height),
	)
}
```

Output inside vim:
> :!'go' 'run' '/code/own/playground/getenv/main.go'  2>&1| tee /tmp/nvim.mafe/TSXPrG/1                                                       
> 2024/11/10 18:34:17 ERROR Current Stdin File Descriptor is no terminal, aborting                                                            
> exit status 1   

Output from within `xterm-kitty` on 13" notebook:

> 💀 94% [18:35:05][ 24.11.0]fsh ♥ go run .
> 2024/11/10 18:35:07 INFO Detected terminal size. columns=140 rows=46
